pkg/pillar/scripts/device-steps.sh

```diff
@@ -375,12 +376,17 @@ if [ ! -s $CONFIGDIR/device.cert.pem ] || [ $RTC = 0 ] || [ -n "$FIRSTBOOT" ]; t
 
     # We need to try our best to setup time *before* we generate the certifiacte.
     # Otherwise the cert may have start date in the future or in 1970
+    # Did NIM get some NTP servers from DHCP? Pick the first one we find
+    ns=$(jq -c .Ports[0].NtpServers < /run/nim/DeviceNetworkStatus/global.json | awk -F\" '{print $2}')
```

Comment:
I would maybe put this command into a function since it repeats 3 times.
But otherwise LGTM.

Reply:
Done. Made that function grab all of the ntp servers (useful down the road) and then pick the first. Thus even if port[1] is the only one with an ntp server it will be picked.